nflow-engine/src/main/java/io/nflow/engine/service/WorkflowDefinitionService.java

```diff
@@ -33,31 +25,17 @@
 
   private static final Logger logger = getLogger(WorkflowDefinitionService.class);
 
-  private AbstractResource nonSpringWorkflowsListing;
   private final Map<String, AbstractWorkflowDefinition<? extends WorkflowState>> workflowDefinitions = new LinkedHashMap<>();
   private final WorkflowDefinitionDao workflowDefinitionDao;
   private final boolean persistWorkflowDefinitions;
+  private final boolean autoInit;
+
 
   @Inject
   public WorkflowDefinitionService(WorkflowDefinitionDao workflowDefinitionDao, Environment env) {
     this.workflowDefinitionDao = workflowDefinitionDao;
     this.persistWorkflowDefinitions = env.getRequiredProperty("nflow.definition.persist", Boolean.class);
-  }
-
-  /**
-   * Add given workflow definitions to the managed definitions.
-   * @param workflowDefinitions The workflow definitions to be added.
-   */
-  @Autowired(required = false)
-  public void setWorkflowDefinitions(Collection<AbstractWorkflowDefinition<? extends WorkflowState>> workflowDefinitions) {
-    for (AbstractWorkflowDefinition<? extends WorkflowState> wd : workflowDefinitions) {
-      addWorkflowDefinition(wd);
-    }
-  }
-
-  @Autowired(required = false)
-  public void setWorkflowDefinitions(@NFlow AbstractResource nflowNonSpringWorkflowsListing) {
-    this.nonSpringWorkflowsListing = nflowNonSpringWorkflowsListing;
+    this.autoInit = env.getRequiredProperty("nflow.autoinit", Boolean.class);
```

Comment:
Could we combine the boolean logic of `autoInit=false` => `persistWorkflowDefinitions=false` already in constructor and not store the autoInit as field?

Reply:
No, since we need to persist in different method depending on autoInit.